dropshot/tests/test_tls.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022 Oxide Computer Company
+/*!
+ * Test cases for TLS support. This validates various behaviors of our TLS mode,
+ * including certificate loading and supported modes.
+ */
+
+use dropshot::ConfigDropshot;
+use dropshot::HttpServerStarter;
+use slog::Logger;
+use std::convert::TryFrom;
+use std::path::Path;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::SystemTime;
+
+pub mod common;
+
+fn create_log(test_name: &str) -> slog::Logger {
+    let log_path = dropshot::test_util::log_file_for_test(test_name)
```

Comment:
I think you want to use `LogContext::new` here.  I'm not sure `log_file_for_test` should even be exposed from test_util.rs.  (Basically, I'm worried these tests leak the log file.)

Reply:
Whoops, copied this from test_config. Thanks! I'll see about fixing things over there, too